### PR TITLE
The great rename

### PR DESCRIPTION
**Pull request by dleehr**
- Project is now D4S2: DukeDS Data Delivery System
- Renames Handover->Delivery, Draft->Share, Reject->Decline
- Stores email templates in the database (none provided)
- Removes and rebuilds migrations (will require fresh database)

### CODE REVIEW

**Comment by dleehr** [comment]
Fixes #52 (it will upon renaming the repo)

**Comment by dleehr** [comment]
@johnbradley I'm assigning this to you for code reviewing purposes. I apologize for mixing substantive changes with the rename, but the new stuff is primarily 

- EmailTemplate and EmailTemplateType in d4s2_api/models.py
- get_X_template_text in switchboard/dds_util.py
- Support for templating subjects and bodies in switchboard/mailer.py

File: d4s2_api/models.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    pass
+
+class EmailTemplateType(models.Model):
+    """
+    Type of email template, e.g. share_project_viewer, delivery, final_notification
+    """
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
+
+    def __str__(self):
+        return self.template_type
```
Comment (johnbradley) [inline d4s2_api/models.py]: Should this be name? It blows up when I try to share a project:
```
AttributeError: 'EmailTemplateType' object has no attribute 'template_type'
```

File: README.md [hunk 2]
```diff
@@ -41,18 +41,18 @@ Installation - Docker Compose
...
 
-3. Edit the `handoverservice.env` file to populate the your DukeDS API details, a django key, and a database username/password to use:
+3. Edit the `d4s2.env` file to populate the your DukeDS API details, a django key, and a database username/password to use:
 
-        HANDOVERSERVICE_SECRET_KEY=some-random-string
-        HANDOVERSERVICE_DDSCLIENT_URL=https://dataservice-host/api/v1
-        HANDOVERSERVICE_DDSCLIENT_AGENT_KEY=your-agent-key
-        POSTGRES_USER=handover_user
+        D4S2_SECRET_KEY=some-random-string
+        D4S2_DDSCLIENT_URL=https://dataservice-host/api/v1
```
Comment (dleehr) [inline README.md]: This is just a sample though though - not meant to be the real URL

File: d4s2_api/models.py (hunk 1 above)
Comment (dleehr) [inline d4s2_api/models.py]: Good catch

**Pull request merged**